lib/rackup/handler/webrick.rb

```diff
@@ -93,18 +93,23 @@ def service(req, res)
 
         status, headers, body = @app.call(env)
         begin
-          res.status = status.to_i
-          io_lambda = nil
+          res.status = status
+
+          if value = headers[RACK_HIJACK]
+            io_lambda = value
+          elsif !body.respond_to?(:to_path) && !body.respond_to?(:each)
+            io_lambda = body
+          end
+
+          if value = headers.delete('set-cookie')
+            res.cookies.concat(Array(value))
+          end
+
           headers.each { |key, value|
-            if key == RACK_HIJACK
```

Comment:
Should we be skipping headers that start with `rack.`?

```suggestion
            next if key.start_with?('rack.')
```

Reply:
According to SPEC, all headers starting with rack must be skipped, so you are correct we should skip them.  I don't think this suggested change is correct, it seems like it would result in invalid syntax.  I'll submit a different fix.

Reply:
Maybe I'm misreading the suggested change.  Maybe your suggested change is only adding the line, and not removing any line, which seems OK.  GitHub's UI could be better here.  Let me try merging your change and see what it does.

Reply:
Looks like I cannot apply the suggested patch:

```
Applying suggestions on deleted lines is currently not supported.
```

Kind of stupid that GitHub would show a UI for a suggestion that is impossible for it to apply.  I'll apply it locally and push a commit for it.

Reply:
Haha, sorry about that.